### ISSUE TITLE
cannot login to Commons

### ISSUE THREAD

**Issue by JiriMatejicek** [yarl/vicuna#165]
For two days, I cannot log in to Commons, Vicuna says 'server error'. However, logging in through a web browser works well. Closing or reinstalling Vicuna 1.23 didn't help.

**Comment by GilPe64** [comment]
Same for me … :-|

**Comment by michal-josef-spacek** [comment]
Which version of Vicuna? You are talking about 1.23. Could you try 1.24?

**Comment by GilPe64** [comment]
Hi,

 

Yes, Im using version 1.23.

 

Until now, I was not able to install version 1.24. The website http://yarlu.pl/vicuna/ only offers 1.23 to download.

 

Checking for updates through the “Help” in Vicuna does neither work …

 



 

GilPe64

**Comment by michal-josef-spacek** [comment]
@GilPe64 https://github.com/yarl/vicuna/releases/tag/1.24

**Comment by JiriMatejicek** [comment]
I've tried version 1.24, but there is no file that would run.

**Comment by michal-josef-spacek** [comment]
@JiriMatejicek Download vicuna-1.24.zip, extract to file system and run bin/vicuna.bat on Windows or bin/vicuna on Linux.

**Comment by GilPe64** [comment]
Sorry, but it is not possible to run the vicuna.bat Windows10. Vers. 21H1

 

GilPe

**Comment by JiriMatejicek** [comment]
Same here. Windows 10, vicuna.bat does not start anything.

**Comment by michal-josef-spacek** [comment]
@JiriMatejicek @GilPe64 And what's a result of vicuna.bat run?

**Comment by GilPe64** [comment]
Hi,

 

A DOS-window appears for half e second, disappears… nothing more.

 

GilPe64

**Comment by michal-josef-spacek** [comment]
@GilPe64 
> Hi, A DOS-window appears for half e second, disappears… nothing more. 

You could run in console or some managar (like FAR manager).
There is some output.

FYI: I tested on Windows 10 and i have issue with bad option --add-opens.  I will look to it, still i don't know how to fix now.
BTW: In 1.24 is fix for some logging issues.

**Comment by JiriMatejicek** [comment]
At first, it looks similar to Windows - just blinks and goes back to FAR manager. Somewehere behind the main window I was able to find this, which may be a clue:

ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
...
ERROR: JAVA_HOME is set to an invalid directory: %JAVA_HOME%                                  
Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.

I have no idea what they mean by 'your environment' and where to set 'JAVA_HOME'. Previous version(s) of Vicuna worked for several years (until few days ago), with no particular relation of Vicuna's and Java's directories.
I have this directory: C:\Program Files\Java with two subfolders, jre1.8.0_191 and jre1.8.0_301 and that's all I needed until now.
Thanks for trying to help.

**Comment by GilPe64** [comment]
Hi,

 

Sorry for not replying earlier, I was away…

 

Trying to start the vicuna.bat in DOS-Windows gives this result:

 



 

GilPe64

**Comment by Silubr1** [comment]
Possibly related to this problem:
https://github.com/ritschwumm/commonist/issues/25

See also https://commons.wikimedia.org/w/index.php?title=Commons:Forum&oldid=597158580#Problem_beim_Hochladen_mit_Commonist
(in German)

**Comment by CodingLama** [comment]
> Possibly related to this problem: [ritschwumm/commonist#25](https://github.com/ritschwumm/commonist/issues/25)
> 
> See also https://commons.wikimedia.org/w/index.php?title=Commons:Forum&oldid=597158580#Problem_beim_Hochladen_mit_Commonist (in German)

Yes, it's related to that problem so it is not a version-specific issue. We had prepared new release of Vicuna and now we are waiting weather someone will fix it.

**Comment by michal-josef-spacek** [comment]
> We had prepared new release of Vicuna and now we are waiting weather someone will fix it.
@CodingLama What do you mean by "We had prepared new release of Vicuna"? Who? Where? Actually last version is 1.24 and no big commits in preparation. Still some issues for next version.

FYI: There are some commits in branch https://github.com/exander77/vicuna , but we need to clean them and i don't know what author of commits mean about moving to main branch.
CC: @exander77

As i understand there are issues with compression and probably in some deprecated api.

**Comment by CodingLama** [comment]
@michal-josef-spacek I don't understand you, but I guess fork of the project does not fix this problem.

**Comment by michal-josef-spacek** [comment]
@CodingLama You are saying "We had prepared new release of Vicuna" and i don't understand what you are saying. 
Last version is https://github.com/yarl/vicuna/releases/tag/1.24
And after 1.24 there are feww small commits now.

**Comment by michal-josef-spacek** [comment]
I tested issue with logging in 1.24 and there is exception with Wiki.java:
java.util.zip.ZipException: Not in GZIP format

So we need to upgrade to new version of Wiki.java

1.23 is obsolete version, because there are other issues with Wiki.java like token (as i remember).

**Comment by michal-josef-spacek** [comment]
Everything which we need is in https://github.com/yarl/vicuna/pull/166 , actually waiting for fix in wiki-java.

**Comment by michal-josef-spacek** [comment]
PR for issue with communication is in https://github.com/yarl/vicuna/pull/172
I created issue for Windows problem: https://github.com/yarl/vicuna/issues/173
